### PR TITLE
[XamlC] don't compile bindings with Source

### LINKED ISSUE #3606: x:DataType tries to force compile bindings which have a Source and Path property set
### Description
Setting the x:DataType of the DataTemplate of a ListView that has a binding to a different object than the current binding context like this
```xml
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:viewModels="clr-namespace:DataTypeSourceBug.ViewModels;assembly=DataTypeSourceBug"
             x:Class="DataTypeSourceBug.MainPage"
             x:Name="Page">
    <ContentPage.BindingContext>
        <viewModels:MainPageViewModel />
    </ContentPage.BindingContext>
    <ListView ItemsSource="{Binding StringList}">
        <ListView.ItemTemplate>
            <DataTemplate x:DataType="x:String">
                <ViewCell>
                    <Label Text="{Binding .}">
                        <Label.GestureRecognizers>
                            <TapGestureRecognizer Command="{Binding Path=BindingContext.InsideListViewCommand,Source={x:Reference Page}}"
                                                  CommandParameter="{Binding .}" />
                        </Label.GestureRecognizers>
                    </Label>
                </ViewCell>
            </DataTemplate>
        </ListView.ItemTemplate>
    </ListView>
</ContentPage>
```
produces an error while compiling with the following stack trace
```
  Resource: DataTypeSourceBug.Views.MainPage.xaml...
   Creating empty MainPage.__InitComponentRuntime ...done.
   Copying body of InitializeComponent to __InitComponentRuntime ...done.
   Parsing Xaml... done.
   Replacing MainPage.InitializeComponent ()... failed.
Views\MainPage.xaml(17,47): error : Position 17:47. Binding: Property 'BindingContext' not found on 'System.String'
   at Xamarin.Forms.Build.Tasks.SetPropertiesVisitor.ParsePath(String path, TypeReference tSourceRef, IXmlLineInfo lineInfo, ModuleDefinition module)
   at Xamarin.Forms.Build.Tasks.SetPropertiesVisitor.<CompileBindingPath>d__30.MoveNext()
   at Xamarin.Forms.Build.Tasks.SetPropertiesVisitor.<ProvideValue>d__29.MoveNext()
   at Xamarin.Forms.Build.Tasks.ILProcessorExtensions.Append(ILProcessor processor, IEnumerable`1 instructions)
   at Xamarin.Forms.Build.Tasks.SetPropertiesVisitor.Visit(ElementNode node, INode parentNode)
   at Xamarin.Forms.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Build.Tasks.SetPropertiesVisitor.SetDataTemplate(IElementNode parentNode, ElementNode node, ILContext parentContext, IXmlLineInfo xmlLineInfo)
   at Xamarin.Forms.Build.Tasks.SetPropertiesVisitor.Visit(ElementNode node, INode parentNode)
   at Xamarin.Forms.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Xaml.RootNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Xamarin.Forms.Build.Tasks.XamlCTask.TryCoreCompile(MethodDefinition initComp, MethodDefinition initCompRuntime, ILRootNode rootnode, Exception& exception)
```
### Steps to Reproduce

1. Try to build the attached repo. 

### Expected Behavior
Compiles fine without trying to compile the bindings that use different Source and Path
### Actual Behavior
Error when compiling
### Basic Information

- Version with issue: 3.1.0
- Last known good version:
- IDE: Visual Studio 15.8
- Platform Target Frameworks: 
  - iOS:
  - Android:
  - UWP:
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Reproduction Link

[DataTypeSourceBug.zip](https://github.com/xamarin/Xamarin.Forms/files/2300162/DataTypeSourceBug.zip)

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

Sources are usually x:reference that can't be resolved at compilation
time.

### Issues Resolved ### 

- fixes #3606

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Build.Tasks/SetPropertiesVisitor.cs [hunk 1]
```diff
@@ -283,7 +283,9 @@ internal static string GetContentProperty(TypeReference typeRef)
 				out markupExtension, out genericArguments))
 			{
 				var acceptEmptyServiceProvider = vardefref.VariableDefinition.VariableType.GetCustomAttribute(module, ("Xamarin.Forms.Core", "Xamarin.Forms.Xaml", "AcceptEmptyServiceProviderAttribute")) != null;
-				if (vardefref.VariableDefinition.VariableType.FullName == "Xamarin.Forms.Xaml.BindingExtension")
+				if (   vardefref.VariableDefinition.VariableType.FullName == "Xamarin.Forms.Xaml.BindingExtension"
```
Comment (adrianknight89) [inline Xamarin.Forms.Build.Tasks/SetPropertiesVisitor.cs]: I noticed this file has many places where unnecessary whitespace is being inserted after the open parenthesis.

Comment (StephaneDelcroix) [reply Xamarin.Forms.Build.Tasks/SetPropertiesVisitor.cs]: that's on purpose, so conditions are indented like this
```
if (   a
    && b)
```
which is more readable (for long and/or complex conditions) than having the operator at the end of the line or at the beginning without indentation
